content/automate/ManageIQ/Infrastructure/VM/Provisioning/Placement.class/__methods__/vmware_best_fit_least_utilized.rb

```diff
@@ -1,37 +1,68 @@
 #
 # Description: This method is used to find all hosts, datastores that are the least utilized
 #
-$evm.log("info", "Args:    #{MIQ_ARGS.inspect}")
-
-# Get variables
-prov = $evm.root["miq_provision"]
-vm = prov.vm_template
-raise "VM not specified" if vm.nil?
-ems = vm.ext_management_system
-raise "EMS not found for VM [#{vm.name}]" if ems.nil?
-
-# Log space required
-$evm.log("info", "vm=[#{vm.name}], space required=[#{vm.provisioned_storage}]")
-
-host = storage = nil
-min_registered_vms = nil
-prov.eligible_hosts.each do |h|
-  next if h.maintenance
-  next unless h.power_state == "on"
-  nvms = h.vms.length
-  if min_registered_vms.nil? || nvms < min_registered_vms
-    storages = h.writable_storages.find_all { |s| s.free_space > vm.provisioned_storage } # Filter out storages that do not have enough free space for the Vm
-    s = storages.max_by(&:free_space)
-    unless s.nil?
-      host    = h
-      storage = s
-      min_registered_vms = nvms
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Provisioning
+          module Placement
+            class VmwareBestFitLeastUtilized
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                @handle.log("info", "Args:    #{MIQ_ARGS.inspect}")
+                best_fit_least_utilized
+              end
+
+              def vm
```

Comment:
@lfu 
A better style might be to use an instance variable like here
https://github.com/ManageIQ/manageiq-content/blob/758b168e4520da52a310bb50666330fae697d2c6/content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/restorevmattributes.rb#L26

This way you can just call vm whenever you need it and if it doesn't exist it will raise the error.

We should do the same for the prov method and object, we could actually call the method provision_request or just request. prov by itself is confusing if its provider or provision request